### DISCUSSION TITLE
Add `detents` prop to `<Modal />`

### DISCUSSION THREAD

**Discussion by nandorojo** [react-native-community/discussions-and-proposals#518] [Ideas]
React Native's `Modal` component should add support for `detents` from iOS.

I'm already using this minimal [patch](https://gist.github.com/barthap/cc8412c5dca6232226c097b368930bba) in my app successfully. 

See [this tweet](https://twitter.com/FernandoTheRojo/status/1575247583634935808?s=20) for a video.

I think the naming should be like so:

```tsx
<Modal 
  presentationStyle="formSheet"
  detents="mediumResizable" // or medium, large
/>
```

Would a PR get merged in React Native to support this feature?

**Comment by thymikee** [comment]
Thanks for posting this! I've seen the tweet and was wondering if there's a corresponding PR :D Proposals that are quite small in scope like this are best when sent as PRs to the react-native repo directly. If there's anything to discuss, like naming, the PR is a proper place to do so.

**Reply by nandorojo** [reply]
Got it. I might not be the best person to submit the PR. I would at-best be copy/pasting code since I don't know Objective C. If someone is willing to pick it up, that would be awesome.

**Reply by thymikee** [reply]
Hah, sounds like a perfect opportunity to try 😃 Maybe the author, @barthap would be interested?

**Reply by barthap** [reply]
I've been planning to open a PR for a while, but my MSc thesis is a priority now. Hopefully I'll have more time next month 🙄

**Reply by thymikee** [reply]
Thanks! 🤞🏼 for your thesis :)

**Reply by hurali97** [reply]
@thymikee I would like to grab this one 🙂

**Reply by barthap** [reply]
@hurali97 Go on! 😉

**Reply by hurali97** [reply]
Fellas a PR is raised [here](https://github.com/facebook/react-native/pull/34834). @barthap and @nandorojo, thanks a bunch for this effort and POC 🌟

**Reply by nandorojo** [reply]
thanks for taking it on! hope to see it get reviewed. although i won't hold my breath...
